### ISSUE TITLE
[quotes] False positive with backtick in declare module

### ISSUE THREAD

**Issue by BBosman** [typescript-eslint/typescript-eslint#757]
**Repro**

```JSON
{
  "rules": {
    "quotes": ["error", "backtick", { "avoidEscape": true }]
  }
}
```

`html.d.ts`:
```TS
declare module '*.html' {
  const value: string;
  export default value;
}
```

**Expected Result**
No rule violations as those can't be backticks.

**Actual Result**
Violation of `quotes`.

**Additional Info**

Actually changing them to backticks gives a compiler error.
```
declare module `*.html` {
  const value: string;
  export default value;
}
```
Error: `Cannot find name 'declare'`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.0` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.10.2` |

**Comment by bradzacher** [comment]
See additional request for quotes rule in: #756

**Comment by a-tarasyuk** [comment]
@bradzacher As I understood, in order to handle that case need to create the new rule `@typescript-eslint/quotes`, add core logic from `quotes` and add additional logic to handle specific cases for TypeScript. Is it true?

**Comment by bradzacher** [comment]
you don't necessarily need to copy the logic from the base rule.
it depends on what the base implementation is exactly.

Sometimes you can get away with adding some logic before just calling the base rule's code.

You can see what I mean in the extension rules we have implemented currently
```
  'camelcase',
  'func-call-spacing',
  'no-array-constructor',
  'no-extra-parens',
  'no-magic-numbers',
  'no-use-before-define',
  'no-useless-constructor',
  'semi',
```

**Comment by a-tarasyuk** [comment]
@bradzacher got it. Thanks :+1:
